src/include/execution/vm/llvm_engine.h

```diff
@@ -190,6 +193,59 @@ class LLVMEngine {
     std::unique_ptr<TPLMemoryManager> memory_manager_;
     std::unordered_map<std::string, void *> functions_;
   };
+
+  // -------------------------------------------------------
+  // LLVM Engine Settings
+  // -------------------------------------------------------
+
+  /**
+   * Engine-wide settings that apply for the entire process in which the LLVM Engine runs.
+   */
+  class Settings {
+   public:
+    /**
+     * Construct a settings instance from the relevant configuration parameters.
+     * @param bytecode_handlers_path The path to the bytecode handlers bitcode file.
+     */
+    explicit Settings(std::string_view bytecode_handlers_path) : bytecode_handlers_path_{bytecode_handlers_path} {}
```

Comment:
If we're going to construct a string anyway, would it make sense to have the caller construct the string and take ownership of it here?

Reply:
I like generally like clean ownership transfers like that, but I don't think this is ideal with the current setup of `DBMain` - we have to declare the default path to the file as a member of `DBMain`, so we would have to move the string out of the `DBMain` instance, which works but seems... wrong. Like we shouldn't be moving members out of `DBMain`. There is the additional consideration that accepting the argument as a `std::string_view` enables us to pass string constants to it, which is nice for the API (I think). Interested to hear your thoughts on those two points.

Reply:
Sounds good to me